ts/src/sourcemapper.ts

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from 'fs';
+import * as _ from 'lodash';
+import * as path from 'path';
+import * as sourceMap from 'source-map';
+const pify = require('pify');
+const pLimit = require('p-limit');
+const readFile = pify(fs.readFile);
+
+const CONCURRENCY = 10;
+const MAP_EXT = '.map';
+
+export interface MapInfoCompiled {
+  mapFile: string;
+  mapConsumer: sourceMap.RawSourceMap;
+}
+
+export interface GeneratedLocation {
+  file: string;
+  name?: string;
+  line: number;
+  column: number;
+}
+
+export interface SourceLocation {
+  file?: string;
+  name?: string;
+  line?: number;
+  column?: number;
+}
+
+/**
+ * @param {!Map} infoMap The map that maps input source files to
+ *  SourceMapConsumer objects that are used to calculate mapping information
+ * @param {string} mapPath The path to the sourcemap file to process.  The
+ *  path should be relative to the process's current working directory
+ * @private
+ */
+async function processSourcemap(
+    infoMap: Map<string, MapInfoCompiled>, mapPath: string) {
+  // this handles the case when the path is undefined, null, or
+  // the empty string
+  if (!mapPath || !_.endsWith(mapPath, MAP_EXT)) {
```

Comment:
Any reason not to use `mapPath.endsWith(MAP_EXT)`? `String.prototype.endsWith` is [available](https://node.green/) in recent Node 6 and we require a minimum of 6.12.x.

You can get rid of the `lodash` dependency.

Reply:
Done.